### PR TITLE
Run HiveMQ & operator as non-root user by default

### PR DESCRIPTION
**Pull request by sbaier1**
still a WIP, dependent images still pending release/update

### CODE REVIEW

File: charts/hivemq-operator/operator-tmpls/cluster-stateful-set.yaml [hunk 1]
```diff
@@ -16,8 +16,14 @@ metadata:
     name: "{{ spec.name }}"
     uid: "{{ spec.metadata.uid }}"
 spec:
-  serviceName: hivemq-{{ spec.name }}-cluster
   replicas: {{ spec.nodeCount }}
+  serviceName: hivemq-{{ spec.name }}-cluster
+  podManagementPolicy: Parallel
+  updateStrategy:
+    type: OnDelete
+  # TODO get this from the CR
```
Comment (schaebo) [inline charts/hivemq-operator/operator-tmpls/cluster-stateful-set.yaml:24]: TODO still valid ?

**Review by schaebo — APPROVED** [review]

File: charts/hivemq-operator/operator-tmpls/cluster-stateful-set.yaml (hunk 1 above)
Comment (patrickjahns) [reply charts/hivemq-operator/operator-tmpls/cluster-stateful-set.yaml:24]: The ToDo is also still in the upstream operator template changes - needs to be removed there when we sync the final template manifestes with the helm chart

File: charts/hivemq-operator/templates/global/hivemq-psp.yaml [hunk 2]
```diff
@@ -31,25 +29,25 @@ spec:
   hostPID: false
   runAsUser:
     # Permits the container to run with root privileges as well.
-    rule: 'RunAsAny'
+    rule: 'MustRunAsNonRoot'
```
Comment (patrickjahns) [inline charts/hivemq-operator/templates/global/hivemq-psp.yaml:32]: The comment above the line would need adjustment, as we are now explicitly forbid to run as root

**Comment by patrickjahns** [comment]
@sbaier1 
Regarding the admission hooks - I believe we need to adjust the `RunAs` configuration as well ( see: https://github.com/hivemq/helm-charts/pull/25/files )

**Pull request merged**